packages/reports/addon/components/filter-builders/date-time.js

```diff
@@ -1,51 +1,178 @@
+import { get, set, computed, action } from '@ember/object';
 import Base from './base';
+import Interval from 'navi-core/utils/classes/interval';
+import Duration from 'navi-core/utils/classes/duration';
+import { getFirstDayOfIsoDateTimePeriod } from 'navi-core/utils/date';
+import moment from 'moment';
+
+export default class extends Base {
+  /**
+   * @property {String} dateTimePeriodName - the date time period
+   */
+  @computed('request.logicalTable.timeGrain')
+  get dateTimePeriodName() {
+    return get(this, 'request.logicalTable.timeGrain.longName');
+  }
 
-export default Base.extend({
   /**
    * @override
-   * @param {Array} supportedOperators - list of valid operators for a date-time filter
+   * @property {Array} supportedOperators - list of valid operators for a date-time filter
    */
-  supportedOperators: computed(function() {
+  @computed('dateTimePeriodName')
+  get supportedOperators() {
     return [
+      {
+        id: 'current',
+        longName: `Current ${this.dateTimePeriodName}`,
+        valuesComponent: 'filter-values/current-period'
+      },
+      {
+        id: 'inPast',
+        longName: 'In The Past',
+        valuesComponent: 'filter-values/lookback-input'
+      },
+      {
+        id: 'since',
+        longName: 'Since',
+        valuesComponent: 'filter-values/since-input'
+      },
       {
         id: 'in',
-        longName: 'In Range',
+        longName: 'Between',
         valuesComponent: 'filter-values/date-range'
+      },
+      {
+        id: 'advanced',
+        longName: 'Advanced',
+        valuesComponent: 'filter-values/advanced-interval-input' // TODO: Show two inputs and range
       }
     ];
-  }),
+  }
 
   /**
    * @property {Object} requestFragment - interval fragment from request
    */
-  requestFragment: undefined,
+  requestFragment = undefined;
+
+  /**
+   * Finds the appropriate interval operator to modify an existing interval
+   * @param {Interval} interval - the interval to choose an operator for
+   */
+  operatorForInterval(interval) {
+    const { start, end } = interval.asStrings();
+
+    const [, /*match*/ lookback /*amount*/, , dateTimePeriodLabel] = /(P)(\d+)(\w)/.exec(start) || [];
+    let operatorId;
+    if (start === 'current' && end === 'next') {
+      operatorId = 'current';
+    } else if (lookback === 'P' && ['D', 'W', 'M', 'Y'].includes(dateTimePeriodLabel) && end === 'current') {
+      operatorId = 'inPast';
+    } else if (moment.isMoment(interval._start) && (end === 'current' || end === 'next')) {
+      operatorId = 'since';
+    } else if (moment.isMoment(interval._start) && moment.isMoment(interval._end)) {
+      operatorId = 'in';
+    } else {
+      operatorId = 'advanced';
+    }
+
+    return this.supportedOperators.find(op => op.id === operatorId);
+  }
+
+  /**
+   *
+   * @param {Interval} interval
+   * @param {String} newOperator
+   */
+  convertIntervalToOperator(interval, dateTimePeriod, newOperator) {
```

Comment:
The name of this function is a little confusing. 

Maybe call it `convertIntervalForNewOperator`?

Reply:
`intervalForOperator`?